starter-code/public/scripts/article.js

```diff
@@ -60,17 +61,40 @@ Article.fetchAll = callback => {
 
 // TODO: Chain together a `map` and a `reduce` call to get a rough count of all words in all articles.
 Article.numWordsAll = () => {
-  return Article.all.map().reduce()
-};
+    return Article.all.map((articles) => {
+      var wordCount =  articles.body.split(' ');
+      return wordCount.length;
+    })
+  .reduce((a, b) => {
+    return a + b;
+    })
+  };
 
 // TODO: Chain together a `map` and a `reduce` call to produce an array of unique author names. You will
 // probably need to use the optional accumulator argument in your reduce call.
 Article.allAuthors = () => {
-  return Article.all.map().reduce();
+  return Article.all.map(authNames=>{
+    return authNames.author;
+  }).reduce((acc, auth) => {
+    return (acc.indexOf(auth) === -1) ? acc.concat([auth]) : acc;
+  },[]);
 };
 
 Article.numWordsByAuthor = () => {
-  return Article.allAuthors().map(author => {
+  return Article.allAuthors().map((author) => {
+      return {
+        name: author,
+        numWords: Article.all.map((article) => {
+          if (article.author === author) {
```

Comment:
Interesting way of doing this. Basically grabbing word counts of articles from specific authors, all without using filter. Great job!